rest/src/main/java/fi/otavanopisto/pyramus/rest/StudentRESTService.java

```diff
@@ -1816,16 +1843,39 @@ public Response updateStudentContactLogEntry(@PathParam("STUDENTID") Long studen
     if (!contactLogEntry.getStudent().getId().equals(contactLogEntry.getStudent().getId())) {
       return Response.status(Status.NOT_FOUND).build();
     }
+    
+    User loggedUser = sessionController.getUser();
+    
+    StaffMember staffMember = userController.findStaffMemberById(loggedUser.getId());
+    
+    if (staffMember == null) {
+      return Response.status(Status.FORBIDDEN).build();
+    }
+    
+    ContactLogAccess access = studentController.resolveContactLogAccess(student);
+    
+    if (access.equals(ContactLogAccess.NONE)) {
+      return Response.status(Status.FORBIDDEN).build();
+    }
+    
+    if (access.equals(ContactLogAccess.OWN)) {
+      if (contactLogEntry.getCreator() != null) {
+        if (!contactLogEntry.getCreator().equals(staffMember)) {
```

Comment:
Not sure if the objects are definitely always the same instance. Safer to compare their ids instead.